request-manager/grapher/grapher.go

```diff
@@ -307,9 +307,27 @@ func (o *Grapher) buildComponent(name string, nodeDefs map[string]*NodeSpec, nod
 		componentsRemaining = len(componentsToAdd)
 	}
 
+	///////////////////////////////////////////////////////////
+	// Third, mark all vertices in graph with start and end sequence ids
+
+	sequenceStartId := g.First.Datum.Id().Id
+	sequenceEndId := g.Last.Datum.Id().Id
+
+	// Update all vertices in this component with Sequence{End,Start}Id.
+	for _, vertex := range g.Vertices {
+		// Vertices will be visited twice. Once when constructing the component
+		// and once when putting the entire graph together. We only want to mark
+		// the Sequence{End,Start}Id when constructing the component.
+		if vertex.SequenceStartId == "" && vertex.SequenceEndId == "" {
+			vertex.SequenceStartId = sequenceStartId
+			vertex.SequenceEndId = sequenceEndId
+		}
+	}
+
 	// Assert g is a well formed graph
 	if !g.IsValidGraph() {
```

Comment:
Request: you can update IsValidGraph() to check that SequenceStartId and SequenceEndId match what we expect them to be. Not a hard blocker, but would be useful if/when we need to update graphing code in the future.